R/Renderer.R

```diff
@@ -286,3 +313,85 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
   lock_objects = TRUE,
   lock_class = TRUE
 )
+
+#' @keywords internal
+block_to_md <- function(block, private, ...) {
+  UseMethod("block_to_md")
+}
+
+#' @method block_to_md default
+#' @keywords internal
+block_to_md.default <- function(block, private, ...) {
+  block
+}
+
+#' @method block_to_md TextBlock
+#' @keywords internal
+block_to_md.TextBlock <- function(block, private, ...) {
+  private$textBlock2md(block)
+}
+
+#' @method block_to_md RcodeBlock
+#' @keywords internal
+block_to_md.RcodeBlock <- function(block, private, ...) {
+  private$rcodeBlock2md(block)
+}
+
+#' @method block_to_md PictureBlock
+#' @keywords internal
+block_to_md.PictureBlock <- function(block, private, ...) {
+  private$pictureBlock2md(block)
+}
+
+#' @method block_to_md TableBlock
+#' @keywords internal
+block_to_md.TableBlock <- function(block, private, ...) {
+  private$tableBlock2md(block)
+}
+
+#' @method block_to_md NewpageBlock
+#' @keywords internal
+block_to_md.NewpageBlock <- function(block, private, ...) {
+  block$get_content()
+}
+
+#' @method block_to_md HTMLBlock
+#' @keywords internal
+block_to_md.HTMLBlock <- function(block, private, ...) {
+  private$htmlBlock2md(block)
+}
+
+#' @method block_to_md character
+#' @keywords internal
+block_to_md.character <- function(block, private, ...) {
+  block
+}
+
+#' @method block_to_md gg
+#' @keywords internal
+block_to_md.gg <- function(block, private, ...) {
+  private$content2md(block)
```

Comment:
`content2md` shouldn't be a part of the renderer class. This violates the encapsulation:
- new class (ReportDocument, gg, rtables, data.frame, ...) are independent so its conversion shouldn't be dependent on the Renderer class or anything else. 
- old classes (Renderer, Previewer) however should be able to call methods to render them to markdown of html.



I see your point in the comment that private$content2md needs to stay. There are numbers of solutions you can use but `private` shouldn't be passed for sure but `output_dir` only.